cosmos/operators/gcp_cloud_run_job.py

```diff
@@ -218,3 +234,73 @@ class DbtCloneGcpCloudRunJobOperator(DbtCloneMixin, DbtGcpCloudRunJobBaseOperato
 
     def __init__(self, *args: Any, **kwargs: Any):
         super().__init__(*args, **kwargs)
+
+
+class DbtWarningGcpCloudRunJobOperator(DbtGcpCloudRunJobBaseOperator, ABC):
+    def __init__(self, on_warning_callback: Callable[..., Any] | None = None, *args: Any, **kwargs: Any) -> None:
+        if not on_warning_callback:
+            super().__init__(*args, **kwargs)
+        else:
+            super().__init__(*args, **kwargs)
+            self.on_warning_callback = on_warning_callback
+
+    def _handle_warnings(self, logs: list[str], context: Context) -> None:
+        """
+         Handles warnings by extracting log issues, creating additional context, and calling the
+         on_warning_callback with the updated context.
+
+        :param logs: The log list with the cleaned Cloud Run Job logs.
+        :param context: The original airflow context in which the build and run command was executed.
+        """
+        test_names, test_results = extract_log_issues(logs)
+
+        warning_context = dict(context)
+        warning_context["test_names"] = test_names
+        warning_context["test_results"] = test_results
+
+        self.on_warning_callback(warning_context)
+
+    def execute(self, context: Context, **kwargs: Any) -> None:
+        result = self.build_and_run_cmd(context=context, cmd_flags=self.add_cmd_flags())
+        log_list = [log for log in result if type(log) == str]  # clean log list with only string type values
+
+        if not (
+            isinstance(context["task_instance"], TaskInstance)
+            and (
+                isinstance(context["task_instance"].task, DbtTestGcpCloudRunJobOperator)
+                or isinstance(context["task_instance"].task, DbtSourceGcpCloudRunJobOperator)
```

Comment:
It feels a bit weird that we check for subclasses here. It feels this logic should be handled directly in the subclasses, not in the parent class